apps/verticalhome/js/stores/item.js

```diff
@@ -339,16 +343,11 @@
               function (txn, store) {
                 var request = store.get(object.index);
                 request.onsuccess = function (req) {
-                  var objFromDB = req.result;
-                  if (objFromDB) {
-                    var DBIndex = objFromDB.index;
-                    objFromDB = object;
-                    objFromDB.index = DBIndex;
-                    store.put(objFromDB);
-                  } else {
-                    store.put(object);
+                  if (req.result) {
+                    object.index = req.result.index;
```

Comment:
I still think, that this code is meaningless, let's discuss about it tomorrow morning.